src/lib/cache.js

```diff
@@ -136,21 +148,21 @@ export default {
     } else {
       data.cached = timestamp
     }
-
-    const payload = zlib.deflateSync(
-                      JSON.stringify(data)
-                    ).toString('base64')
     /* eslint-enable no-param-reassign */
 
-    return client.set(
-      key,
-      payload,
-      "EX",
-      CACHE_LIFETIME_IN_SECONDS,
-      err => {
-        if (err) error(err)
-      }
-    )
+    return deflateP(data).then(deflatedData => {
+      const payload = deflatedData.toString('base64')
+      verbose(`CACHE SET: ${key}: ${payload}`)
+      return client.set(
+        key,
+        payload,
+        "EX",
+        CACHE_LIFETIME_IN_SECONDS,
+        err => {
+          if (err) error(err)
+        }
+      )
+    })
```

Comment:
Seeing as the caller (at least currently) is not going to handle rejections of the returned promise, maybe you should add a `.chain(error)` here so it at least gets logged in a proper way?

Reply:
Sure - any ideas about waiting for this in the context of the api `loader()`?  I still can't get this last test to pass ^

Reply:
Or if we are happy that `cache.set`s now happen async, maybe we just drop this test?

Reply:
How about mocking `cache.set(…)` instead of verifying the data has been set? https://facebook.github.io/jest/docs/en/mock-functions.html